### PR TITLE
api-docs: expand intro and drop curl url environment variable

### PR DESCRIPTION
**Pull request by pinheadmz**
I added a long intro to the API docs covering most of the FAQs we get from new developers using the hsd API for the first time.

In addition, I replaced all the curl $url stuff with the explicit path. It's too hard to find what $url stores in the docs, and it makes the examples impossible to copy-paste unless you've already set the environment variable

### CODE REVIEW

File: src/api-docs/source/index.html.md [hunk 1]
```diff
@@ -45,18 +45,127 @@ search: true
...
-The default hsd HTTP server listens on port (`12037` for main, `13037` for testnet, `14037` for regtest, and `15037` for simnet). It exposes a REST JSON, as well as a JSON-RPC API.
+hsd is a fork of [bcoin](https://github.com/bcoin-org/bcoin) and so if you are familiar with it,
+you'll find that the API is very similar to the [bcoin API](http://bcoin.io/api-docs/).
+bcoin was released with a JSON-RPC API that was based on the Bitcoin Core RPC interface.
+The actual Bitcoin Core RPC API has changed a lot since then, but if you are familiar with
+using bitcoind, you'll find a lot of the same RPC calls available in hsd.
+
+**There are four API servers in hsd**
+
+The full node and the wallet are separate processes in hsd, and their API servers run on separate ports.
```
Comment (tynes) [inline src/api-docs/source/index.html.md:56]: Nit: they can be separate processes but do not need to be

File: src/api-docs/source/index.html.md [hunk 2]
```diff
@@ -45,18 +45,127 @@ search: true
...
+
+Network   | Wallet API Port | Node API Port
+--------- | --------------- | -------------
+main      | 12039           | 12037
+testnet   | 13039           | 13037
+regtest   | 14039           | 14037
+simnet    | 15039           | 15037
+
+In addition, each server has two API formats: JSON-RPC and a RESTful HTTP API.
+In general, the HTTP APIs respond to `GET` requests and the RPC APIs respond to
```
Comment (tynes) [inline src/api-docs/source/index.html.md:66]: Nit: I don't think we need this sentence `In general...`
There are a bunch of wallet HTTP endpoints that use `POST` and the RPC API has examples below.

Comment (tynes) [reply src/api-docs/source/index.html.md:66]: Something that might be more useful to state:

Much of the functionality is duplicated between the HTTP API and RPC API. Note that if multiple user-agents are using the wallet at the same time, the wallet RPC API is not safe as it maintains global state. It is recommended to use the wallet HTTP API in multiprocess/multithreaded environments.

File: src/api-docs/source/index.html.md [hunk 3]
```diff
@@ -45,18 +45,127 @@ search: true
...
+**Get a [transaction by hash](#get-tx-by-txhash) from the full node HTTP API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14037/tx/4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f
+</code>
+
+**Get a [raw transaction by hash](#getrawtransaction) from the full node RPC API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14037
```
Comment (tynes) [inline src/api-docs/source/index.html.md:80]: Missing newline escape characters `\`

File: src/api-docs/source/index.html.md [hunk 4]
```diff
@@ -45,18 +45,127 @@ search: true
...
+</code>
+
+**Indexing**
+
+hsd has two indexer modules that are inactive by default. When turned on, the indexers
+enable additional API calls.
+
+Transaction indexer: enabled by `--index-tx=true`. Allows lookup of arbitrary transactions
+by their hash (txid). When disabled, the wallet still tracks all of its own transactions
+by hash and the full node can still be used to look up UTXO (unspent transaction outputs).
```
Comment (tynes) [inline src/api-docs/source/index.html.md:112]: The UTXOs are indexed by outpoint, which is the transaction ID that created the UTXO along with its index in that transaction.

File: src/api-docs/source/index.html.md [hunk 5]
```diff
@@ -45,18 +45,127 @@ search: true
...
+
+Address indexer: enabled by `--index-address=true`. Allows lookup of all transactions
+involving a certain address.
+
+
+**Wallet: BIP44**
+
+The hsd wallet follows a [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki)
+structure with cointype `5353` for mainnet. A single hsd instance can have multiple BIP44
```
Comment (tynes) [inline src/api-docs/source/index.html.md:121]: Nit: `5353h`

File: src/api-docs/source/index.html.md [hunk 6]
```diff
@@ -45,18 +45,127 @@ search: true
...
+wallets in its database.
+
+The hierarchy looks like this:
+
+`WalletDB`: A levelDB database stored in a `/wallet` directory, accessed by the wallet module
```
Comment (tynes) [inline src/api-docs/source/index.html.md:126]: Its `~/.hsd/wallet` right?

File: src/api-docs/source/index.html.md [hunk 7]
```diff
@@ -45,18 +45,127 @@ search: true
...
+plugged-in by default to the full node. Can process transactions for multiple wallets.
+
+`Wallet`: A BIP44 wallet based on 128 or 256 bits of entropy and backed up by a 12- or 24-
+word phrase. Multiple accounts can be derived from a single wallet.
+
+`Account`: A branch of a BIP44 wallet. Multiple addresses can be derived from a single
+account, and the wallet can be configured to spend coins from a single specified account.
+
+`Address`: A hash of a public key or script that can be given to a counterparty with the intent
+of receiving funds.
```
Comment (tynes) [inline src/api-docs/source/index.html.md:136]: There are two types of outputs defined for version 0 outputs. Pay-to-Witness-Pubkeyhash have a 20 byte hash of a public key and can be spent if the owner reveals the public key and provides a signature from the corresponding private key. Pay-to-Witness-Scripthash have a 32 byte hash of a Bitcoin script and can be spent if the owner reveals the script along with any additional data required to satisfy the script.

File: src/api-docs/source/index.html.md (hunk 3 above)
Comment (pinheadmz) [reply src/api-docs/source/index.html.md:80]: After struggling with a few approaches to this, I just moved the entire example section to the right side where "code" is displayed and added the newline characters.

File: src/api-docs/source/index.html.md (hunk 5 above)
Comment (pinheadmz) [reply src/api-docs/source/index.html.md:121]: hmmmmmmmmm. I think BIP44 hardens cointype by definition, I just want to indicate the number as it appears in https://github.com/satoshilabs/slips/blob/master/slip-0044.md

File: src/api-docs/source/index.html.md (hunk 6 above)
Comment (pinheadmz) [reply src/api-docs/source/index.html.md:126]: I'll remove the slash so it doesnt look like a path, I think thats out of context.

**Comment by pinheadmz** [comment]
Thanks @tynes ! Addressed everything and pushed a commit.

**Pull request merged**